substrate-node/pallets/pallet-smart-contract/src/types.rs

```diff
@@ -251,3 +252,38 @@ pub struct Provider<AccountId> {
     pub who: AccountId,
     pub take: u8,
 }
+
+// metadata length limited to 64 bytes (2 public keys)
+pub const MAX_METADATA_LENGTH: u32 = 64;
+
+#[derive(Clone, Eq, PartialEq, RuntimeDebugNoBound, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[scale_info(skip_type_params(T))]
+#[codec(mel_bound())]
+pub struct ServiceContract<T: Config> {
+    pub service_twin_id: u32,
+    pub consumer_twin_id: u32,
+    pub base_fee: u64,
+    pub variable_fee: u64,
+    pub metadata: BoundedVec<u8, ConstU32<MAX_METADATA_LENGTH>>,
+    pub accepted_by_service: bool,
+    pub accepted_by_consumer: bool,
+    pub last_bill: u64,
+    pub state: ServiceContractState,
+    pub phantom: PhantomData<T>,
```

Comment:
Looks like you don't need `<T: Config>` here, so suggest to remove the generic param + phantomdata

Reply:
I was looking for a way to get rid of it but it is required here
https://github.com/threefoldtech/tfchain/blob/6be200760b0265630d8dc6867a1d8624bf28613a/substrate-node/pallets/pallet-smart-contract/src/cost.rs#L100

Reply:
After some tries, I did not find a way to remove `<T>`